api/v1/fluxreport_types.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2024 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package v1
+
+import (
+	"strings"
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const FluxReportKind = "FluxReport"
+
+// FluxReportSpec defines the observed state of a Flux installation.
+type FluxReportSpec struct {
+	// Distribution is the version information of the Flux installation.
+	// +required
+	Distribution FluxDistributionStatus `json:"distribution"`
+
+	// ComponentsStatus is the status of the Flux controller deployments.
+	// +optional
+	ComponentsStatus []FluxComponentStatus `json:"components,omitempty"`
+
+	// ReconcilersStatus is the list of Flux reconcilers and
+	// their statistics grouped by API kind.
+	// +optional
+	ReconcilersStatus []FluxReconcilerStatus `json:"reconcilers,omitempty"`
+
+	// SyncStatus is the status of the cluster sync
+	// Source and Kustomization resources.
+	// +optional
+	SyncStatus *FluxSyncStatus `json:"sync,omitempty"`
+}
+
+// FluxDistributionStatus defines the version information of the Flux instance.
+type FluxDistributionStatus struct {
+	// Entitlement is the entitlement verification status.
+	// +required
+	Entitlement string `json:"entitlement"`
+
+	// Status is the status of the Flux instance.
+	// +required
+	Status string `json:"status"`
+
+	// Version is the version of the Flux instance.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// ManagedBy is the name of the operator managing the Flux instance.
+	// +optional
+	ManagedBy string `json:"managedBy,omitempty"`
+}
+
+// FluxComponentStatus defines the observed state of a Flux component.
+type FluxComponentStatus struct {
+	// Name is the name of the Flux component.
+	// +required
+	Name string `json:"name"`
+
+	// Ready is the readiness status of the Flux component.
+	// +required
+	Ready bool `json:"ready"`
```

Comment:
Having `ready` and `status` is confusing I think. Perhaps `ready` and `message`?